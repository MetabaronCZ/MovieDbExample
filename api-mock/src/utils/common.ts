import { Request } from 'express';

export type QueryParam = Request['query'][number];
export type Query = Record<string, QueryParam>;

const wait = (delay: number): Promise<void> => {
  return new Promise((resolve) => setTimeout(resolve, delay));
};

export const requestDelay = (cb: () => void): void => {
  const delay = 800 + Math.floor(400 * Math.random());
  wait(delay)
    .then(cb)
    .catch((error: unknown) => {
      console.error(error);
    });
};

// extract simple string value from Express query parameter result
export const parseQueryParam = (param: QueryParam): string => {
  const value = Array.isArray(param) ? param[0] : param;
  return 'string' === typeof value ? value : '';
};

type Entries<T extends Record<string, unknown>> = {
  readonly [K in keyof T]: [K, T[K]];
}[keyof T][];

// extract object entries, correctly typed
export const getObjectEntries = <T extends Record<string, unknown>>(
  obj: T,
): Entries<T> => {
  return Object.entries(obj) as Entries<T>;
};
