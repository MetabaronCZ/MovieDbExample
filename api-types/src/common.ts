export const perPages = [25, 50, 100] as const;
export const defaultPerPage = perPages[0];

export const sortDirections = ['ascending', 'descending'] as const;
export type SortDirection = (typeof sortDirections)[number];
