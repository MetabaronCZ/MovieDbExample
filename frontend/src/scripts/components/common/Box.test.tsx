import { getByText } from '@testing-library/dom';
import { describe, expect, it } from '@jest/globals';

import { Box } from 'components/common/Box';
import { TestComponentWrapper, waitForComponent } from 'test-utils/component';

describe('components/common/Box', () => {
  it('should render properly', async () => {
    const { container } = await waitForComponent(<Box>CONTENT</Box>, {
      wrapper: TestComponentWrapper,
    });
    expect(container).toMatchSnapshot();

    const content = getByText(container, 'CONTENT');
    expect(content).toBeInTheDocument();
  });
});
