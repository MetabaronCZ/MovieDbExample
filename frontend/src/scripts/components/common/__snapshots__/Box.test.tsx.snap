// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Box should render properly 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  gap: 16px;
}

.c1 {
  width: 100%;
  padding: 16px;
  background: #fff;
  box-shadow: 1px 1px 4px #dae0e3;
}

<div>
  <div
    class="c0 c1"
  >
    CONTENT
  </div>
</div>
`;
